fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACAuthorizingRealm.java

```diff
@@ -231,25 +235,22 @@ public boolean supports(final AuthenticationToken token) {
     }
 
     private FedoraResource getResourceOrParentFromPath(final FedoraId fedoraId) {
-        FedoraResource resource = null;
-        log.debug("Attempting to get FedoraResource for {}", fedoraId.getFullIdPath());
         try {
-            resource = resource(fedoraId);
-            log.debug("Got FedoraResource for {}", fedoraId.getFullIdPath());
-        } catch (final PathNotFoundException e) {
-            log.debug("Path {} does not exist", fedoraId.getFullIdPath());
-            // go up the path looking for a node that exists
-            if (fedoraId.getFullId().length() > 1) {
-                final int lastSlash = fedoraId.getFullId().lastIndexOf("/");
-                final int end = lastSlash > 0 ? lastSlash : lastSlash + 1;
-                resource = getResourceOrParentFromPath(FedoraId.create(fedoraId.getFullId().substring(0, end)));
+            log.debug("Testing FedoraResource for {}", fedoraId.getFullIdPath());
+            return this.resourceFactory.getResource(transaction(), fedoraId);
+        } catch (final PathNotFoundException exc) {
+            log.debug("Resource {} not found getting container", fedoraId.getFullIdPath());
+            final String txID = transaction() == null ? null : transaction().getId();
```

Comment:
Not something that should be addressed here, but I've seen this transaction code everywhere. Perhaps it would be cleaner if `transaction()` returned `Optional<Transaction>`?

Reply:
Not sure I understand what you are thinking? Because the transactionManager returns a transaction object or null, wouldn't getting an Optional<Transaction> just mean a different test?

Reply:
Yes, and no. I don't use optional a lot because I think Java's implementation has some flaws, but it can be useful in some cases, especially when you want to remind the consumer of an API that a method could legitimately return null. This forces the proper handling of a null condition, which is easy to forget to do.

In this particular case, the code could be rewritten as, if `transaction` returned `Optional`:

```
final String txID = transaction().map(Transaction::getId).orElse(null);
```

Whether or not you think that's better or worse than a null check is largely a matter of taste. Regardless, I just mentioned it because it was on my mind, and not because I am expecting you to change anything here.